interfaces/builtin/optical_drive_test.go

```diff
@@ -77,20 +103,61 @@ func (s *OpticalDriveInterfaceSuite) TestSanitizeSlot(c *C) {
 }
 
 func (s *OpticalDriveInterfaceSuite) TestSanitizePlug(c *C) {
-	c.Assert(interfaces.BeforePreparePlug(s.iface, s.plugInfo), IsNil)
+	c.Assert(interfaces.BeforePreparePlug(s.iface, s.testPlugDefaultInfo), IsNil)
+	c.Assert(interfaces.BeforePreparePlug(s.iface, s.testPlugReadonlyInfo), IsNil)
+	c.Assert(interfaces.BeforePreparePlug(s.iface, s.testPlugWritableInfo), IsNil)
 }
 
 func (s *OpticalDriveInterfaceSuite) TestAppArmorSpec(c *C) {
-	spec := &apparmor.Specification{}
-	c.Assert(spec.AddConnectedPlug(s.iface, s.plug, s.slot), IsNil)
-	c.Assert(spec.SecurityTags(), DeepEquals, []string{"snap.consumer.app"})
-	c.Assert(spec.SnippetForTag("snap.consumer.app"), testutil.Contains, `/dev/sr[0-9]* r,`)
+	type options struct {
+		appName         string
+		includeSnippets []string
+		excludeSnippets []string
+	}
+	checkConnectedPlugSnippet := func(plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot, opts *options) {
+		apparmorSpec := &apparmor.Specification{}
+		err := apparmorSpec.AddConnectedPlug(s.iface, plug, slot)
+		c.Assert(err, IsNil)
+		c.Assert(apparmorSpec.SecurityTags(), DeepEquals, []string{opts.appName})
+		for _, expectedSnippet := range opts.includeSnippets {
+			c.Assert(apparmorSpec.SnippetForTag(opts.appName), testutil.Contains, expectedSnippet)
+		}
+		for _, unexpectedSnippet := range opts.excludeSnippets {
+			c.Assert(apparmorSpec.SnippetForTag(opts.appName), Not(testutil.Contains), unexpectedSnippet)
+		}
+	}
+
+	expectedSnippet1 := `/dev/scd[0-9]* r,`
+	expectedSnippet2 := `/dev/scd[0-9]* w,`
+
+	checkConnectedPlugSnippet(s.testPlugDefault, s.slot, &options{
+		appName:         "snap.consumer.app",
+		includeSnippets: []string{expectedSnippet1},
+		excludeSnippets: []string{expectedSnippet2},
+	})
+	checkConnectedPlugSnippet(s.testPlugReadonly, s.slot, &options{
+		appName:         "snap.consumer.app-readonly",
+		includeSnippets: []string{expectedSnippet1},
+		excludeSnippets: []string{expectedSnippet2},
+	})
+	checkConnectedPlugSnippet(s.testPlugWritable, s.slot, &options{
+		appName:         "snap.consumer.app-writable",
+		includeSnippets: []string{expectedSnippet1, expectedSnippet2},
+		excludeSnippets: []string{},
+	})
+	checkConnectedPlugSnippet(s.testPlugWritable, s.slot, &options{
+		appName:         "snap.consumer.app-writable",
+		includeSnippets: []string{expectedSnippet1, expectedSnippet2},
+		excludeSnippets: []string{},
+	})
```

Comment:
Thanks for these the positive and negative tests! You are doing the 'testPlugWritable' test twice though.

Reply:
oops. That's fixed now.